cgi-bin/DW/Console/Command/BonusIcons.pm

```diff
@@ -56,25 +59,54 @@ sub execute {
     }
 
     return $self->error( 'Invalid subcommand.' )
-        if $cmd && $cmd !~ /^(?:add|remove)$/;
+        if $cmd && $cmd !~ /^(?:add|remove|xfer)$/;
 
-    return $self->error( 'Count must be a positive integer.' )
-        unless $count =~ /^\d+$/;
-    $count += 0;
+    if ( $cmd eq 'add' || $cmd eq 'remove' ) {
 
-    if ( $cmd eq 'add' ) {
-        my $new = max( $to_u->prop( 'bonus_icons' ) + $count, 0 );
-        $to_u->set_prop( bonus_icons => $new );
-        LJ::statushistory_add( $to_u, $remote, 'bonus_icons',
-                sprintf( 'Added %d icons, new total: %d.', $count, $new ) );
-        $self->print( sprintf( 'User now has %d icons.', $new ) );
+        my $count = $cmdarg;
+
+        return $self->error( 'Count must be a positive integer.' )
+            unless $count =~ /^\d+$/;
+        $count += 0;
+
+        if ( $cmd eq 'add' ) {
+            my $new = max( $to_u->prop( 'bonus_icons' ) + $count, 0 );
+            $to_u->set_prop( bonus_icons => $new );
+            LJ::statushistory_add( $to_u, $remote, 'bonus_icons',
+                    sprintf( 'Added %d icons, new total: %d.', $count, $new ) );
+            $self->print( sprintf( 'User now has %d icons.', $new ) );
+
+        } elsif ( $cmd eq 'remove' ) {
+            my $new = max( $to_u->prop( 'bonus_icons' ) - $count, 0 );
+            $to_u->set_prop( bonus_icons => $new );
+            LJ::statushistory_add( $to_u, $remote, 'bonus_icons',
+                    sprintf( 'Removed %d icons, new total: %d.', $count, $new ) );
+            $self->print( sprintf( 'User now has %d icons.', $new ) );
+
+        }
+
+    } elsif ( $cmd eq 'xfer' ) {
+        my $destination_u = LJ::load_user( $cmdarg );
+
+        return $self->error( 'Invalid target user.' )
+            unless $destination_u;
+        return $self->error( 'E-mail addresses do not match.' )
+            unless $to_u->email_raw eq $destination_u->email_raw; 
+        return $self->error( 'One or more email address(es) not confirmed.' )
+            unless $to_u->is_validated && $destination_u->is_validated;
 
-    } elsif ( $cmd eq 'remove' ) {
-        my $new = max( $to_u->prop( 'bonus_icons' ) - $count, 0 );
-        $to_u->set_prop( bonus_icons => $new );
+        my $xfer_count = $to_u->prop( 'bonus_icons' );
+        $to_u->set_prop( bonus_icons => 0 );
         LJ::statushistory_add( $to_u, $remote, 'bonus_icons',
-                sprintf( 'Removed %d icons, new total: %d.', $count, $new ) );
-        $self->print( sprintf( 'User now has %d icons.', $new ) );
+                sprintf( 'Transferred %d icons to %s.', $xfer_count,
+                    $destination_u->user ) );
+        my $new_total = $destination_u->prop( 'bonus_ions' ) + $xfer_count;
```

Comment:
As amusing as the idea is of catalyzing a chemical reaction, I think you should probably correct this typo. :wink: 

And yeah, go ahead and squish the commits if you like.